### DISCUSSION TITLE
null value in column id of relation "users"

### DISCUSSION THREAD

**Discussion by nicitaacom** [supabase/supabase#16485] [Questions]
Please help me with this error

```
RegisterPage.tsx:80 register -  {code: '23502', details: 'Failing row contains 
(2023-08-12 18:38:28.913696+00, , , null).', hint: null, message: 'null value in
 column "id" of relation "users" violates not-null constraint'}code: "23502"details:
 "Failing row contains (2023-08-12 18:38:28.913696+00, , , null)."hint: 
nullmessage: "null value in column \"id\" of relation \"users\" violates not-null 
constraint"[[Prototype]]: Object register @ RegisterPage.tsx:80
await in register (async)
```

### My register function that trows this error in "catch"
```tsx
 async function register(e: React.FormEvent) {
    e.preventDefault()
    try {
      if (email && password) {
        const response = await supabase.auth.signUp({ email: email, password: password })
        if (response.error) throw response.error
        if (response.data.user?.id) {
          const { error } = await supabase.from("users")
            .insert({ username: username })
          if (error) throw error
          console.log('52')
          setIsError(false)
          setIsSuccess(true)
          /* auth user after success register */
          const response = await supabase.auth.signInWithPassword({ email: email, password: password })

          const setProfilePictureUrl = async () => {
            const { data } = await supabase.from("users").select("profile_picture_url").eq("id", response.data.user?.id)
            if (data) {
              ``
              userStore.setProfilePictureUrl(data[0]?.profile_picture_url as string)
            }
          }

          const { data } = await supabase.from("users")
            .select("username")
            .eq("id", response.data.user?.id)
          if (data && data.length > 0 && data[0]?.username) {
            const { username } = data[0]
            navigate('/', { replace: true })
          }

          setProfilePictureUrl()
        }
      }
    } catch (error) {
      console.error("register - ", error)
      setIsSuccess(false)
      setIsError(true)
    }
  }
  ```
  
  ### Schema public - table "users"
  
![image](https://github.com/supabase/supabase/assets/39565703/0c29ead1-282c-4db9-ae51-e00cbc28993b)
![image](https://github.com/supabase/supabase/assets/39565703/de89e5d9-db15-4020-843a-8b4125cf7386)

I suposse that its because I have 2 settings here (but shown 1)
![image](https://github.com/supabase/supabase/assets/39565703/9d8e800a-fb5b-46b4-9569-f366949ac9b0)

  
  If you need additional information feel free to ask

**Comment by GaryAustin1** [comment]
You don't provide any value for id on your insert.  You either need a default value in your table or a value on your insert.  If that is a signed in user doing the insert then you can use auth.uid() as the default in your table setup.   fk relations do not populate automatically.

**Reply by GaryAustin1** [reply]
You should not add new answers, but reply to any answers you get.

Your error probably means you don't have a signed in user in the Supabase client when you make the call.   Auth.uid() will be null in that case.

**Reply by GaryAustin1** [reply]
Also if you are using email confirm, the signUp does not return with a session (just tentative user info).
So after `const response = await supabase.auth.signUp({ email: email, password: password })` you cannot make supabase calls as the user after this call, unless you have confirm email off in your dashboard email provider settings.

Have you looked at the user management starter used in the guides or SQL editor welcome section?
With that you can include your username as part of signUp and then the trigger function will add that to your public table.

**Reply by nicitaacom** [reply]
Do I need add new row username in schema "auth" table "users"?
![image](https://github.com/supabase/supabase/assets/39565703/b5b7c667-bc23-4cdc-b7f1-a48ea1ed6160)

**Comment by nicitaacom** [comment]
If you mean default value in table "users" in column "id" - I did like like so
![image](https://github.com/supabase/supabase/assets/39565703/0b545f3b-e5be-4d99-9c4b-e13f3ef1aebe)

And got this error
![image](https://github.com/supabase/supabase/assets/39565703/9f8f8189-0a8a-4a52-afca-83ec5dbcb1d2)

```
code: '23502', details: 'Failing row contains (2023-08-13 07:16:13.407727+00, , , null).',
 hint: null, message: 'null value in column "id" of relation "users" violates not-null constraint'}
code
: 
"23502"
details
: 
"Failing row contains (2023-08-13 07:16:13.407727+00, , , null)."
hint
: 
null
message
: 
"null value in column \"id\" of relation \"users\" violates not-null constraint"
[[Prototype]]
: 
Object
```

**Comment by nicitaacom** [comment]
# Resolved
I fixed this error
Error was in my code
```tsx
 async function register(e: React.FormEvent) {
    e.preventDefault()
    try {
      if (email && password) {
        const response = await supabase.auth.signUp({ email: email, password: password })
        if (response.error) throw response.error
        if (response.data.user?.id) {
          const { error } = await supabase.from("users")
            .insert({ id: response.data.user.id, username: usernameValue })
          if (error) throw error
          setIsError(false)
          setIsSuccess(true)
        }
      }
    } catch (error) {
      console.error("register - ", error)
      setIsSuccess(false)
      setIsError(true)
    }
  }
  ```
  
 Previosly I didn't set id here
 ```tsx
 .insert({ id: response.data.user.id, username: usernameValue })
 ```
 
 I thought that id takes form schama "auth" table "users"
